controllers/storageclaim_controller.go

```diff
@@ -398,15 +404,27 @@ func (r *StorageClaimReconciler) reconcilePhases() (reconcile.Result, error) {
 				// storageclaim is clusterscoped resources using its
 				// hash as the clusterID
 				data["clusterID"] = r.storageClaimHash
-				if resource.Name == "cephfs" {
-					volumeSnapshotClass = r.getCephFSVolumeSnapshotClass(data)
-				} else if resource.Name == "ceph-rbd" {
-					volumeSnapshotClass = r.getCephRBDVolumeSnapshotClass(data)
-				}
+				labels["ramendr.openshift.io/storageID"] = storageID
+				volumeSnapshotClass = r.getCephDriverVolumeSnapshotClass(resource.Name, data)
+				utils.AddLabels(volumeSnapshotClass, labels)
 				utils.AddAnnotation(volumeSnapshotClass, storageClaimAnnotation, r.storageClaim.Name)
 				if err := r.createOrReplaceVolumeSnapshotClass(volumeSnapshotClass); err != nil {
 					return reconcile.Result{}, fmt.Errorf("failed to create or update VolumeSnapshotClass: %s", err)
 				}
+			case "VolumeGroupSnapshotClass":
+				var volumeGroupSnapshotClass *groupsnapapi.VolumeGroupSnapshotClass
+				data["csi.storage.k8s.io/group-snapshotter-secret-namespace"] = r.OperatorNamespace
+				// generate a new clusterID for cephfs subvolumegroup, as
+				// storageclaim is clusterscoped resources using its
+				// hash as the clusterID
+				data["clusterID"] = r.storageClaimHash
```

Comment:
@raaizik I am confused.
 If we can decide on a cluster ID locally why do we need to send it from the provider (as implemented in this PR: https://github.com/red-hat-storage/ocs-operator/pull/2661)?

Reply:
As are we ➡️ [discussion](https://github.com/red-hat-storage/ocs-operator/pull/2661#discussion_r1642481467)